api/src/pcapi/core/categories/subcategories_v2.py

```diff
@@ -2,19 +2,497 @@
+from pcapi.domain.movie_types import MOVIE_TYPES
 from pcapi.domain.music_types import MUSIC_TYPES_LABEL_BY_CODE
 from pcapi.domain.music_types import MusicType
-from pcapi.domain.music_types import music_types
+from pcapi.domain.music_types import MUSIC_TYPES
 from pcapi.domain.show_types import SHOW_TYPES_LABEL_BY_CODE
 from pcapi.domain.show_types import ShowType
-from pcapi.domain.show_types import show_types
+from pcapi.domain.show_types import SHOW_TYPES
+
+
+class SearchNode:
+    def __init__(
+        self,
+        label: str,
+        parents: list[str] | None = None,
+        technical_name: str | None = None,
+        gtls: list[str] | None = None,
+        position: int | None = None,
+        search_filter: str | None = None,
+    ) -> None:
+        self.id = technical_name or str(uuid.uuid4())
+        self.label = label
+        self.gtls = gtls
+        self.parents = parents or []
+        self.position = position
+        self.search_filter = search_filter
+
+    @property
+    def search_value(self) -> str | None:
+        return None
+
+
+class SearchGroup(SearchNode):
+    search_filter: str = "searchGroups"
+
+    def __new__(cls, *args: typing.Any, **kwargs: typing.Any) -> "SearchGroup":
+        obj = super().__new__(cls)
+        if "instances" not in cls.__dict__:
+            cls.instances = []
+
+        cls.instances.append(obj)
+        return obj
+
+    @property
+    def search_value(self) -> str | None:
+        return self.id
+
+    @property
+    def name(self) -> str:  # For retro-compatibility with route /subcategories_v2
+        return self.id
+
+    @property
+    def value(self) -> str:  # For retro-compatibility with route /subcategories_v2
+        return self.label
+
+
+class NativeCategory(SearchNode):
+    search_filter: str = "nativeCategoryId"
+
+    def __new__(cls, *args: typing.Any, **kwargs: typing.Any) -> "NativeCategory":
+        obj = super().__new__(cls)
+        if "instances" not in cls.__dict__:
+            cls.instances = []
+
+        cls.instances.append(obj)
+        return obj
+
+    @property
+    def search_value(self) -> str | None:
+        return self.id
+
+    @property
+    def name(self) -> str:  # For retro-compatibility with route /subcategories_v2
+        return self.id
+
+    @property
+    def value(self) -> str:  # For retro-compatibility with route /subcategories_v2
+        return self.label
+
+
+class BookGenre(SearchNode): ...
+
+
+class MovieGenre(SearchNode):
+    search_filter: str = "movieGenres"
+
+    @property
+    def search_value(self) -> str | None:
+        return self.id
+
+
+class MusicGenre(SearchNode):
+    search_filter: str = "musicType"
+
+    @property
+    def search_value(self) -> str | None:
+        return self.label
+
+
+class ShowGenre(SearchNode):
+    search_filter: str = "showType"
+
+    @property
+    def search_value(self) -> str | None:
+        return self.label
+
+
+def book_nodes() -> list[BookGenre]:
```

Comment:
```suggestion
def get_book_nodes() -> list[BookGenre]:
```
pareil pour les autres fonctions qui récupèrent une liste, c'est surtout important ici parce `book` veut dire à la fois livre et réserver